### PR TITLE
Refactor and cleanup logging

### PR DESCRIPTION
**Pull request by ying-zheng**
1. Remove MA UUID and UnifiedLog
MA UUID is a uuid used for tracing requests across multiple services and stages
UnifiedLog is a log tag to mark log messages that are useful for Michelangelo users and log messages that are only for Michelangelo developers.
Both were not fully implemented. Based on our experience in the last 2 years, these features are not very useful and hard to maintain.

2. Remove the tally and log tags for user email, caller, source
This information is added to yarpc context by Uber's internal libraries. We can't get this info in open source code. We will retrieve this information from context in Uber's internal implementation of audit logger and metrics emit functions

3. Rename tally sub scope name from 'uapi_yarpc' to 'ma_api'
In OSS Michelangelo we don't use the term "unified API", as there is only one Michelanelo API

4. Rename tally tag name from "project" to "namespace"
Although, in most cases, k8s namespaces corresponds to michelangelo projects. There are a few cases the namespaces are not projects. Calling this tag "namespace" is more accurate

5. Do not truncate JSON string in the generated yarpc service code
Different logging services have different maximum string length limits. We will do the truncation in the custom implementations of audit log and log sink, rater than setting a global limit.

6. Remove the fields of AuditLogEvent that are not set / used in OSS michelangelo

7. Remove "Unified Log"
We used to use 2 loggers: zap sugar logger and logr logger. "Unified Log" was added as an abstract layer on top of these 2 loggers.  Now, we should only use logr. "Unified Log" is no longer needed

7. added go/vendor in .bazelignore
This directory is generated by 'go mod vendor' command

### CODE REVIEW

File: .bazelignore [hunk 1]
```diff
@@ -1 +1,2 @@
 pkg
+go/vendor
```
Comment (linzhp) [inline .bazelignore:2]: how is this populated?

File: go/logging/audit_log_struct.go [hunk 2]
```diff
@@ -5,16 +5,8 @@ type AuditLogEvent struct {
...
-	Source     string
-	Headers    string
 	Request    string
 	Response   string
 	Error      string
-	MAUUID     string
-	RuntimeEnv string
-	CRDDiff    string
 	PreCRD     string
-	TimeStamp  int64
```
Comment (sallycr) [inline go/logging/audit_log_struct.go:18]: Isn't this timestamp useful?

File: go/logging/constants.go [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package logging
+
+// MichelangeloAPIScopeName is the metrics sub-scope name for all Michelangelo API calls
+const MichelangeloAPIScopeName = "ma_api"
```
Comment (sallycr) [inline go/logging/constants.go]: can we just use 'api'?

File: go/logging/audit_log_struct.go (hunk 2 above)
Comment (sallycr) [reply go/logging/audit_log_struct.go:18]: hm. this was not used before.

Comment (ying-zheng) [reply go/logging/audit_log_struct.go:18]: Some of these fields were from heatpipe schema. We don't set these fields in our code

Comment (sallycr) [reply go/logging/audit_log_struct.go:18]: got it.

**Review by sallycr — APPROVED** [review]
Thanks for refactoring for logging.

**Comment by sallycr** [comment]
@ying-zheng thanks for the description of the pull reqs. It helps to review the diff.

**Pull request merged**